### PR TITLE
Kythant/undocked winrt activation/apartments tests

### PR DESCRIPTION
**Pull request by kythant**
Test infrastructure and tests for apartment activation.

### CODE REVIEW

**Comment by kythant** [comment]
I think I need to convert my tabs into spaces I will do that in another PR once this is in.

File: src/winrt/activation/catalog.cpp [hunk 1]
```diff
@@ -30,21 +30,29 @@ struct component
...
-	ABI::Windows::Foundation::ThreadingType threading_model;
+    ABI::Windows::Foundation::ThreadingType threading_model;
 
-    bool LoadModule() 
+    HRESULT LoadModule()
     {
         if (handle == nullptr)
         {
             handle = LoadLibraryW(module_path.c_str());
+            if (handle == nullptr)
```
Comment (manodasanW) [inline src/winrt/activation/catalog.cpp]: Any reason not to use RETURN_IF_WIN32_ERROR macros?

File: src/winrt/activation/catalog.cpp [hunk 2]
```diff
@@ -30,21 +30,29 @@ struct component
...
+                return HRESULT_FROM_WIN32(GetLastError());;
+            }
             this->get_activation_factory = (activation_factory_type)GetProcAddress(handle, "DllGetActivationFactory");
+            if (this->get_activation_factory == nullptr)
+            {
+                return HRESULT_FROM_WIN32(GetLastError());;
+            }
         }
-        return handle != nullptr && this->get_activation_factory != nullptr;
+        return (handle != nullptr && this->get_activation_factory != nullptr) ? S_OK : E_FAIL;
```
[the remark this answers is not in the document]
Comment (kythant) [reply src/winrt/activation/catalog.cpp]: It is kinda

[the remark this answers is not in the document]
Comment (kythant) [reply src/winrt/activation/catalog.cpp]: I think this should be an OR

File: src/winrt/activation/catalog.cpp (hunk 1 above)
Comment (kythant) [reply src/winrt/activation/catalog.cpp]: Can't depend on WIL

**Pull request merged**